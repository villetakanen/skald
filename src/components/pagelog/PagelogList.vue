<template>
  <v-card class="sidebar">
    <v-card-title>{{$t('pagelog.card-title')}}</v-card-title>
    <v-card-text>
      <PagelogItem
        :siteid="'siteid'"
        :count="11"/>
    </v-card-text>
  </v-card>
</template>

<script lang="ts">
import Vue from 'vue'
import VueCompositionApi, { defineComponent, ref } from '@vue/composition-api'
import PagelogItem from './PagelogItem.vue'
Vue.use(VueCompositionApi)

export default defineComponent({
  components: {
    PagelogItem
  },
  props: {
    siteid: {
      type: [String],
      required: false
    }
  },
  setup (props) {
    const siteid = ref('')
    if (props.siteid) siteid.value = props.siteid
    return { siteid }
  }
})
</script>
