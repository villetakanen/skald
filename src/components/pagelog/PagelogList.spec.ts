import { shallowMount } from '@vue/test-utils'
import PagelogList from './PagelogList.vue'

describe('Mounted App', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(PagelogList, {
      mocks: {
        $t: () => {}
      },
      propsData: {
        siteid: 'skald'
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
