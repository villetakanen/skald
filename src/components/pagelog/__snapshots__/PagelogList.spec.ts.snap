// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mounted App renders correctly 1`] = `
<v-card-stub
  class="sidebar"
  loaderheight="4"
  tag="div"
>
  <v-card-title-stub>
    
  </v-card-title-stub>
   
  <v-card-text-stub>
    <pagelogitem-stub
      count="11"
      siteid="siteid"
    />
  </v-card-text-stub>
</v-card-stub>
`;
